README.md
# Free-Voice

Free Voice est une micro-entreprise qui n'avait pas de site web. Afin de la faire connaître en sus des réseaux sociaux, nous lui en faisons un.


PHPMailer/changelog.txt
ChangeLog
  
NOTE: THIS VERSION OF PHPMAILER IS DESIGNED FOR PHP5/PHP6.
      IT WILL NOT WORK WITH PHP4.
  
Version 5.2.4 (February 19, 2013)
  * Fix tag and version bug.
  * un-deprecate isSMTP(), isMail(), IsSendmail() and
    isQmail().

Version 5.2.3 (February 8, 2013)
  * Fix issue with older PCREs and ValidateAddress() (Bugz: 124)

Version 5.2.2 (December 3, 2012)
  * Some fixes and syncs from https://github.com/Synchro/PHPMailer

Version 5.2.2-rc2 (November 6, 2012)
  * Fix SMTP server rotation (Bugz: 118)
  * Allow override of autogen'ed 'Date' header (for Drupal's
    og_mailinglist module)
  * No whitespace after '-f' option (Bugz: 116)
  * Work around potential warning (Bugz: 114)

Version 5.2.2-rc1 (September 28, 2012)
  * Header encoding works with long lines (Bugz: 93)
  * Turkish language update (Bugz: 94)
  * undefined $pattern in EncodeQ bug squashed (Bugz: 98)
  * use of mail() in safe_mode now works (Bugz: 96)
  * ValidateAddress() now 'public static' so people can override the
    default and use their own validation scheme.
  * ValidateAddress() no longer uses broken FILTER_VALIDATE_EMAIL
  * Added in AUTH PLAIN SMTP authentication

Version 5.2.2-beta2 (August 17, 2012)
  * Fixed Postfix VERP support (Bugz: 92)
  * Allow action_function callbacks to pass/use
    the From address (passed as final param)
  * Prevent inf look for get_lines() (Bugz: 77)
  * New public var ($UseSendmailOptions). Only pass sendmail()
    options iff we really are using sendmail or something sendmail
    compatible. (Bugz: 75)
  * default setting for LE returned to "\n" due to popular demand.

Version 5.2.2-beta1 (July 13, 2012)
  * Expose PreSend() and PostSend() as public methods to allow
    for more control if serializing message sending.
  * GetSentMIMEMessage() only constructs the message copy when
    needed. Save memory.
  * Only pass params to mail() if the underlying MTA is
    "sendmail" (as defined as "having the string sendmail
    in its pathname") [#69]
  * Attachments now work with Amazon SES and others [Bugz#70]
  * Debug output now sent to stdout (via echo) or error_log [Bugz#5]
  * New var: Debugoutput (for above) [Bugz#5]
  * SMTP reads now Timeout aware (new var: Timeout=15) [Bugz#71]
  * SMTP reads now can have a Timelimit associated with them
    (new var: Timelimit=30)[Bugz#71]
  * Fix quoting issue associated with charsets
  * default setting for LE is now RFC compliant: "\r\n"
  * Return-Path can now be user defined (new var: ReturnPath)
    (the default is "" which implies no change from previous
    behavior, which was to use either From or Sender) [Bugz#46]
  * X-Mailer header can now be disabled (by setting to a
    whitespace string, eg "  ") [Bugz#66]
  * Bugz closed: #68, #60, #42, #43, #59, #55, #66, #48, #49,
                 #52, #31, #41, #5. #70, #69
  
Version 5.2.1 (January 16, 2012)
  * Closed several bugs#5
  * Performance improvements
  * MsgHTML() now returns the message as required.
  * New method: GetSentMIMEMessage() (returns full copy of sent message)
  
Version 5.2 (July 19, 2011)
  * protected MIME body and header
  * better DKIM DNS Resource Record support
  * better aly handling
  * htmlfilter class added to extras
  * moved to Apache Extras
  
Version 5.1 (October 20, 2009)
  * fixed filename issue with AddStringAttachment (thanks to Tony)
  * fixed "SingleTo" property, now works with Senmail, Qmail, and SMTP in
    addition to PHP mail()
  * added DKIM digital signing functionality
    New properties:
    - DKIM_domain (sets the domain name)
    - DKIM_private (holds DKIM private key)
    - DKIM_passphrase (holds your DKIM passphrase)
    - DKIM_selector (holds the DKIM "selector")
    - DKIM_identity (holds the identifying email address)
  * added callback function support
    - callback function parameters include:
      result, to, cc, bcc, subject and body
    * see the test/test_callback.php file for usage.
  * added "auto" identity functionality
    - can automatically add:
      - Return-path (if Sender not set)
      - Reply-To (if ReplyTo not set)
    - can be disabled:
      - $mail->SetFrom('yourname@yourdomain.com','First Last',false);
      - or by adding the $mail->Sender and/or $mail->ReplyTo properties
    Note: "auto" identity added to help with emails ending up in spam
    or junk boxes because of missing headers
  
Version 5.0.2 (May 24, 2009)
  * Fix for missing attachments when inline graphics are present
  * Fix for missing Cc in header when using SMTP (mail was sent,
    but not displayed in header -- Cc receiver only saw email To: 
    line and no Cc line, but did get the email (To receiver
    saw same)
  
Version 5.0.1 (April 05, 2009)
  * Temporary fix for missing attachments
  
Version 5.0.0 (April 02, 2009)
  
  * With the release of this version, we are initiating a new version numbering
    system to differentiate from the PHP4 version of PHPMailer.
  * Most notable in this release is fully object oriented code.
  class.smtp.php:
  * Refactored class.smtp.php to support new exception handling
    code size reduced from 29.2 Kb to 25.6 Kb
  * Removed unnecessary functions from class.smtp.php:
    public function Expand($name) {
    public function Help($keyword="") {
    public function Noop() {
    public function Send($from) {
    public function SendOrMail($from) {
    public function Verify($name) {
  class.phpmailer.php:
  * Refactored class.phpmailer.php with new exception handling
  * Changed processing functionality of Sendmail and Qmail so they cannot be
    inadvertently used
  * removed getFile() function, just became a simple wrapper for
    file_get_contents()
  * added check for PHP version (will gracefully exit if not at least PHP 5.0)
  class.phpmailer.php enhancements
  * enhanced code to check if an attachment source is the same as an embedded or
    inline graphic source to eliminate duplicate attachments
  New /test_script

PHPMailer/test/test_callback.php
<html>
<head>
<title>PHPMailer Lite - DKIM and Callback Function test</title>
</head>
<body>

<?php
/* This is a sample callback function for PHPMailer Lite.
 * This callback function will echo the results of PHPMailer processing.
 */

/* Callback (action) function
 *   bool    $result        result of the send action
 *   string  $to            email address of the recipient
 *   string  $cc            cc email addresses
 *   string  $bcc           bcc email addresses
 *   string  $subject       the subject
 *   string  $body          the email body
 * @return boolean
 */
function callbackAction ($result, $to, $cc, $bcc, $subject, $body) {
  /*
  this callback example echos the results to the screen - implement to
  post to databases, build CSV log files, etc., with minor changes
  */
  $to  = cleanEmails($to,'to');
  $cc  = cleanEmails($cc[0],'cc');
  $bcc = cleanEmails($bcc[0],'cc');
  echo $result . "\tTo: "  . $to['Name'] . "\tTo: "  . $to['Email'] . "\tCc: "  . $cc['Name'] . "\tCc: "  . $cc['Email'] . "\tBcc: "  . $bcc['Name'] . "\tBcc: "  . $bcc['Email'] . "\t"  . $subject . "<br />\n";
  return true;
}

$testLite = false;

if ($testLite) {
  require_once '../../PHPMailer_5.2.4/class.phpmailer-lite.php';
  $mail = new PHPMailerLite();
} else {
  require_once '../class.phpmailer.php';
  $mail = new PHPMailer();
}

try {
  $mail->IsMail(); // telling the class to use SMTP
  $mail->SetFrom('you@yourdomain.com', 'Your Name');
  $mail->AddAddress('another@yourdomain.com', 'John Doe');
  $mail->Subject = 'PHPMailer Lite Test Subject via Mail()';
  $mail->AltBody = 'To view the message, please use an HTML compatible email viewer!'; // optional - MsgHTML will create an alternate automatically
  $mail->MsgHTML(file_get_contents('contents.html'));
  $mail->AddAttachment('images/phpmailer.gif');      // attachment
  $mail->AddAttachment('images/phpmailer_mini.gif'); // attachment
  $mail->action_function = 'callbackAction';
  $mail->Send();
  echo "Message Sent OK</p>\n";
} catch (phpmailerException $e) {
  echo $e->errorMessage(); //Pretty error messages from PHPMailer
} catch (Exception $e) {
  echo $e->getMessage(); //Boring error messages from anything else!
}

function cleanEmails($str,$type) {
  if ($type == 'cc') {
    $addy['Email'] = $str[0];
    $addy['Name']  = $str[1];
    return $addy;
  }
  if (!strstr($str, ' <')) {
    $addy['Name']  = '';
    $addy['Email'] = $addy;
    return $addy;
  }
  $addyArr = explode(' <', $str);
  if (substr($addyArr[1],-1) == '>') {
    $addyArr[1] = substr($addyArr[1],0,-1);
  }
  $addy['Name']  = $addyArr[0];
  $addy['Email'] = $addyArr[1];
  $addy['Email'] = str_replace('@', '&#64;', $addy['Email']);
  return $addy;
}

?>
</body>
</html>


PHPMailer/examples/test_smtp_basic.php
<html>
<head>
<title>PHPMailer - SMTP basic test with authentication</title>
</head>
<body>

<?php

//error_reporting(E_ALL);
error_reporting(E_STRICT);

date_default_timezone_set('America/Toronto');

require_once('../class.phpmailer.php');
//include("../../PHPMailer_5.2.4/examples/class.smtp.php"); // optional, gets called from within class.phpmailer.php if not already loaded

$mail             = new PHPMailer();

$body             = file_get_contents('contents.html');
$body             = preg_replace('/[\]/','',$body);

$mail->IsSMTP(); // telling the class to use SMTP
$mail->Host       = "mail.yourdomain.com"; // SMTP server
$mail->SMTPDebug  = 2;                     // enables SMTP debug information (for testing)
                                           // 1 = errors and messages
                                           // 2 = messages only
$mail->SMTPAuth   = true;                  // enable SMTP authentication
$mail->Host       = "mail.yourdomain.com"; // sets the SMTP server
$mail->Port       = 26;                    // set the SMTP port for the GMAIL server
$mail->Username   = "yourname@yourdomain"; // SMTP account username
$mail->Password   = "yourpassword";        // SMTP account password

$mail->SetFrom('name@yourdomain.com', 'First Last');

$mail->AddReplyTo("name@yourdomain.com","First Last");

$mail->Subject    = "PHPMailer Test Subject via smtp, basic with authentication";

$mail->AltBody    = "To view the message, please use an HTML compatible email viewer!"; // optional, comment out and test

$mail->MsgHTML($body);

$address = "whoto@otherdomain.com";
$mail->AddAddress($address, "John Doe");

$mail->AddAttachment("images/phpmailer.gif");      // attachment
$mail->AddAttachment("images/phpmailer_mini.gif"); // attachment

if(!$mail->Send()) {
  echo "Mailer Error: " . $mail->ErrorInfo;
} else {
  echo "Message sent!";
}

?>

</body>
</html>


PHPMailer/test_script/index.php
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<title>PHPMailer Test Page</title>
<script type="text/javascript" src="scripts/shCore.js"></script>
<script type="text/javascript" src="scripts/shBrushPhp.js"></script>
<link type="text/css" rel="stylesheet" href="styles/shCore.css"/>
<link type="text/css" rel="stylesheet" href="styles/shThemeDefault.css"/>
<script type="text/javascript">
  SyntaxHighlighter.config.clipboardSwf = 'scripts/clipboard.swf';
  SyntaxHighlighter.all();
</script>
</head>
<body >
<?php

echo 'Current PHP version: ' . phpversion() . "<br />";

if ( substr(phpversion(),0,1) < 5 ) { echo exit("ERROR: Wrong PHP version"); }

$CFG['smtp_debug']        = 1;
$CFG['smtp_server']       = 'mail.yourserver.com';
$CFG['smtp_port']         = '25';
$CFG['smtp_authenticate'] = 'true';
$CFG['smtp_username']     = 'name@yourserver.com';
$CFG['smtp_password']     = 'yourpassword';

if ( $_POST['submit'] == "Submit" ) {

  class phpmailerAppException extends Exception {
    public function errorMessage() {
      $errorMsg = '<strong>' . $this->getMessage() . "</strong><br />";
      return $errorMsg;
    }
  }

  try {
    $to = $_POST['To_Email'];
    if(filter_var($to, FILTER_VALIDATE_EMAIL) === FALSE) {
      throw new phpmailerAppException("Email address " . $to . " is invalid -- aborting!<br />");
    }
  } catch (phpmailerAppException $e) {
    echo $e->errorMessage();
    return false;
  }

  require_once("../class.phpmailer.php");

  $mail = new PHPMailer();

  if ( $_POST['Message'] == '' ) {
    $body             = file_get_contents('contents.html');
  } else {
    $body = $_POST['Message'];
  }

  if ( $_POST['test_type'] == "smtp" ) {
    $mail->IsSMTP();  // telling the class to use SMTP
    $mail->SMTPDebug  = $_POST['smtp_debug'];
    $mail->SMTPAuth   = $_POST['smtp_authenticate'];     // enable SMTP authentication
    $mail->Port       = $_POST['smtp_port'];             // set the SMTP port
    $mail->Host       = $_POST['smtp_server'];           // SMTP server
    $mail->Username   = $_POST['authenticate_username']; // SMTP account username
    $mail->Password   = $_POST['authenticate_password']; // SMTP account password
  } elseif ( $_POST['test_type'] == "mail" ) {
    $mail->IsMail();      // telling the class to use PHP's Mail()
  } elseif ( $_POST['test_type'] == "sendmail" ) {
    $mail->IsSendmail();  // telling the class to use Sendmail
  } elseif ( $_POST['test_type'] == "qmail" ) {
    $mail->IsQmail();     // telling the class to use Qmail
  }

  if ( $_POST['From_Name'] != '' ) {
    $mail->AddReplyTo($_POST['From_Email'],$_POST['From_Name']);
    $mail->From       = $_POST['From_Email'];
    $mail->FromName   = $_POST['From_Name'];
  } else {
    $mail->AddReplyTo($_POST['From_Email']);
    $mail->From       = $_POST['From_Email'];
    $mail->FromName   = $_POST['From_Email'];
  }

  if ( $_POST['To_Name'] != '' ) {
    $mail->AddAddress($to,$_POST['To_Name']);
  } else {
    $mail->AddAddress($to);
  }

  if ( $_POST['bcc_Email'] != '' ) {
    $indiBCC = explode(" ", $_POST['bcc_Email']);
    foreach ($indiBCC as $key => $value) {
      $mail->AddBCC($value);
    }
  }

  if ( $_POST['cc_Email'] != '' ) {
    $indiCC = explode(" ", $_POST['cc_Email']);
    foreach ($indiCC as $key => $value) {
      $mail->AddCC($value);
    }
  }

  $mail->Subject  = $_POST['Subject'] . ' (PHPMailer test using ' . strtoupper($_POST['test_type']) . ')';

  // below can be found at http://www.chuggnutt.com/html2text
  //  bundled in ./extras/
  require_once('../../PHPMailer_5.2.4/class.html2text.inc');
  $h2t =& new html2text($body);
  $mail->AltBody = $h2t->get_text();
  //$mail->AltBody    = "To view the message, please use an HTML compatible email viewer!"; // optional, comment out and test
  $mail->WordWrap   = 80; // set word wrap

  $mail->MsgHTML($body);

  // $mail->IsHTML(true); // send as HTML

  $mail->AddAttachment("images/aikido.gif", "aikido.gif");  // optional name
  $mail->AddAttachment("images/phpmailer.gif", "phpmailer.gif");  // optional name

  try {
    if ( !$mail->Send() ) {
      $error = "Unable to send to: " . $to . "<br />";
      throw new phpmailerAppException($error);
    } else {
      echo 'Message has been sent using ' . strtoupper($_POST['test_type']) . "<br /><br />";
    }
  }
  catch (phpmailerAppException $e) {
    $errorMsg[] = $e->errorMessage();
  }

  if ( count($errorMsg) > 0 ) {
    foreach ($errorMsg as $key => $value) {
      $thisError = $key + 1;
      echo $thisError . ': ' . $value;
    }
  }
  ?>
  <form method="POST" enctype="multipart/form-data">
  <?php $value = ( $_POST['From_Name'] != '' ) ? $_POST['From_Name'] : ''; ?>
  <input type="hidden" name="From_Name" value="<?php echo $value; ?>">
  <?php $value = ( $_POST['From_Email'] != '' ) ? $_POST['From_Email'] : ''; ?>
  <input type="hidden" name="From_Email" value="<?php echo $value; ?>">
  <?php $value = ( $_POST['To_Name'] != '' ) ? $_POST['To_Name'] : ''; ?>
  <input type="hidden" name="To_Name" value="<?php echo $value; ?>">
  <?php $value = ( $_POST['To_Email'] != '' ) ? $_POST['To_Email'] : ''; ?>
  <input type="hidden" name="To_Email" value="<?php echo $value; ?>">
  <?php $value = ( $_POST['cc_Email'] != '' ) ? $_POST['cc_Email'] : ''; ?>
  <input type="hidden" name="cc_Email" value="<?php echo $value; ?>">
  <?php $value = ( $_POST['bcc_Email'] != '' ) ? $_POST['bcc_Email'] : ''; ?>
  <input type="hidden" name="bcc_Email" value="<?php echo $value; ?>">
  <?php $value = ( $_POST['Subject'] != '' ) ? $_POST['Subject'] : ''; ?>
  <input type="hidden" name="Subject" value="<?php echo $value; ?>">
  <?php $value = ( $_POST['Message'] != '' ) ? $_POST['Message'] : ''; ?>
  <input type="hidden" name="Message" value="<?php echo $value; ?>">
  <?php $value = ( $_POST['test_type'] != '' ) ? $_POST['test_type'] : 'mail'; ?>
  <input type="hidden" name="test_type" value="<?php echo $value; ?>">
  <?php $value = ( $_POST['smtp_debug'] != '' ) ? $_POST['smtp_debug'] : $CFG['smtp_debug']; ?>
  <input type="hidden" name="smtp_debug" value="<?php echo $value; ?>">
  <?php $value = ( $_POST['smtp_server'] != '' ) ? $_POST['smtp_server'] : $CFG['smtp_server']; ?>
  <input type="hidden" name="smtp_server" value="<?php echo $value; ?>">
  <?php $value = ( $_POST['smtp_port'] != '' ) ? $_POST['smtp_port'] : $CFG['smtp_port']; ?>
  <input type="hidden" name="smtp_port" value="<?php echo $value; ?>">
  <?php $value = ( $_POST['smtp_authenticate'] != '' ) ? $_POST['smtp_authenticate'] : $CFG['smtp_authenticate']; ?>
  <input type="hidden" name="smtp_authenticate" value="<?php echo $value; ?>">
  <?php $value = ( $_POST['authenticate_username'] != '' ) ? $_POST['authenticate_username'] : $CFG['smtp_username']; ?>
  <input type="hidden" name="authenticate_username" value="<?php echo $value; ?>">
  <?php $value = ( $_POST['authenticate_password'] != '' ) ? $_POST['authenticate_password'] : $CFG['smtp_password']; ?>
  <input type="hidden" name="authenticate_password" value="<?php echo $value; ?>">
  <input type="submit" value="Start Over" name="submit">
  </form><br />
  <br />
  Script:<br />
<pre class="brush: php;">
class phpmailerAppException extends Exception {
  public function errorMessage() {
    $errorMsg = '<strong>' . $this->getMessage() . "</strong><br />";
    return $errorMsg;
  }
}

try {
  $to = <?php echo $_POST['To_Email']; ?>;
  if(filter_var($to, FILTER_VALIDATE_EMAIL) === FALSE) {
    throw new phpmailerAppException("Email address " . $to . " is invalid -- aborting!<br />");
  }
} catch (phpmailerAppException $e) {
  echo $e->errorMessage();
  return false;
}

require_once("../class.phpmailer.php");

$mail = new PHPMailer();

<?php
if ( $_POST['Message'] == '' ) {
  echo '$body             = file_get_contents(\'contents.html\');' . "\n";
} else {
  echo '$body = ' . $_POST['Message'] . "\n";
}

echo "\n";

if ( $_POST['test_type'] == "smtp" ) {
  echo '$mail->IsSMTP();  // telling the class to use SMTP' . "\n";
  echo '$mail->SMTPDebug  = ' . $_POST['smtp_debug'] . "\n";
  echo '$mail->SMTPAuth   = ' . $_POST['smtp_authenticate'];     // enable SMTP authentication' . "\n";
  echo '$mail->Port       = ' . $_POST['smtp_port'];             // set the SMTP port' . "\n";
  echo '$mail->Host       = ' . $_POST['smtp_server'];           // SMTP server' . "\n";
  echo '$mail->Username   = ' . $_POST['authenticate_username']; // SMTP account username' . "\n";
  echo '$mail->Password   = ' . $_POST['authenticate_password']; // SMTP account password' . "\n";
} elseif ( $_POST['test_type'] == "mail" ) {
  echo '$mail->IsMail();      // telling the class to use PHP\'s Mail()' . "\n";
} elseif ( $_POST['test_type'] == "sendmail" ) {
  echo '$mail->IsSendmail();  // telling the class to use Sendmail' . "\n";
} elseif ( $_POST['test_type'] == "qmail" ) {
  echo '$mail->IsQmail();     // telling the class to use Qmail' . "\n";
}
?>

$mail->AddReplyTo('<?php echo $_POST['From_Email']; ?>','<?php echo $_POST['From_Name']; ?>');

$mail->From       = '<?php echo $_POST['From_Email']; ?>';
$mail->FromName   = '<?php echo $_POST['From_Name']; ?>';

<?php
if ( $_POST['To_Name'] != '' ) {
  ?>
$mail->AddAddress('<?php echo $to; ?>','<?php echo $_POST['To_Name']; ?>');
  <?php
} else {
  ?>
$mail->AddAddress('<?php echo $to; ?>');
  <?php
}
if ( $_POST['bcc_Email'] != '' ) {
  $indiBCC = explode(" ", $_POST['bcc_Email']);
  foreach ($indiBCC as $key => $value) {
echo '$mail->AddBCC(\'' . $value . '\');<br />';
  }
}

if ( $_POST['cc_Email'] != '' ) {
  $indiCC = explode(" ", $_POST['cc_Email']);
  foreach ($indiCC as $key => $value) {
echo '$mail->AddCC(\'' . $value . '\');<br />';
  }
}
?>

$mail->Subject  = <?php echo $_POST['Subject']; ?> (PHPMailer test using <?php echo strtoupper($_POST['test_type']); ?>)

require_once('../class.html2text.inc');
$h2t =& new html2text($body);
$mail->AltBody = $h2t->get_text();
$mail->WordWrap   = 80; // set word wrap

$mail->MsgHTML($body);

$mail->AddAttachment("images/aikido.gif", "aikido.gif");  // optional name
$mail->AddAttachment("images/phpmailer.gif", "phpmailer.gif");  // optional name

try {
  if ( !$mail->Send() ) {
    $error = "Unable to send to: " . $to . "<br />";
    throw new phpmailerAppException($error);
  } else {
    echo 'Message has been sent using <?php echo strtoupper($_POST['test_type']); ?><br /><br />';
  }
} catch (phpmailerAppException $e) {
  $errorMsg[] = $e->errorMessage();
}

if ( count($errorMsg) > 0 ) {
  foreach ($errorMsg as $key => $value) {
    $thisError = $key + 1;
    echo $thisError . ': ' . $value;
  }
}
</pre>



  <?php
} else {
  ?>
  <style>
  body {
    font-family: Arial, Helvetica, Sans-Serif;
    font-size: 11px;
  }
  td {
    font-size: 11px;
  }
  td.colleft {
    align: right;
    text-align: right;
    width: 30%;
  }
  td.colrite {
    text-align: left;
    width: 70%;
  }
  </style>
  <form method="POST" enctype="multipart/form-data">
  <table border="1" width="900" cellspacing="0" cellpadding="5" style="border-collapse: collapse" bgcolor="#C0C0C0">
    <tr>
      <td valign="top";><strong>Message</strong><br /><br />
        <table border="1" width="450" cellspacing="0" cellpadding="5" style="border-collapse: collapse;" bgcolor="#FFFFFF">
          <tr>
            <td class="colleft">From Name</td>
            <?php $value = ( $_POST['From_Name'] != '' ) ? $_POST['From_Name'] : ''; ?>
            <td class="colrite"><input type="text" name="From_Name" value="<?php echo $value; ?>" style="width:99%;"></td>
          </tr>
          <tr>
            <td class="colleft">From Email Address</td>
            <?php $value = ( $_POST['From_Email'] != '' ) ? $_POST['From_Email'] : ''; ?>
            <td class="colrite"><input type="text" name="From_Email" value="<?php echo $value; ?>" style="width:99%;"></td>
          </tr>
          <tr>
            <td class="colleft">To Name</td>
            <?php $value = ( $_POST['To_Name'] != '' ) ? $_POST['To_Name'] : ''; ?>
            <td class="colrite"><input type="text" name="To_Name" value="<?php echo $value; ?>" style="width:99%;"></td>
          </tr>
          <tr>
            <td class="colleft">To Email Address</td>
            <?php $value = ( $_POST['To_Email'] != '' ) ? $_POST['To_Email'] : ''; ?>
            <td class="colrite"><input type="text" name="To_Email" value="<?php echo $value; ?>" style="width:99%;"></td>
          </tr>
          <tr>
            <td class="colleft">cc Email Addresses <small>(separate with commas)</small></td>
            <?php $value = ( $_POST['cc_Email'] != '' ) ? $_POST['cc_Email'] : ''; ?>
            <td class="colrite"><input type="text" name="cc_Email" value="<?php echo $value; ?>" style="width:99%;"></td>
          </tr>
          <tr>
            <td class="colleft">bcc Email Addresses <small>(separate with commas)</small></td>
            <?php $value = ( $_POST['bcc_Email'] != '' ) ? $_POST['bcc_Email'] : ''; ?>
            <td class="colrite"><input type="text" name="bcc_Email" value="<?php echo $value; ?>" style="width:99%;"></td>
          </tr>
          <tr>
            <td class="colleft">Subject</td>
            <?php $value = ( $_POST['Subject'] != '' ) ? $_POST['Subject'] : ''; ?>
            <td class="colrite"><input type="text" name="Subject" value="<?php echo $value; ?>" style="width:99%;"></td>
          </tr>
          <tr>
            <td class="colleft">Message<br /><small>If blank, will use content.html</small></td>
            <?php $value = ( $_POST['Message'] != '' ) ? $_POST['Message'] : ''; ?>
            <td class="colrite"><textarea name="Message" style="width:99%;height:50px;"><?php echo $value; ?></textarea></td>
          </tr>
        </table>
      </td>
      <td valign="top"><strong>Mail Test Specs</strong><br /><br />
        <table border="1" width="450" cellspacing="0" cellpadding="5" style="border-collapse: collapse;" bgcolor="#FFFFFF">
          <tr>
            <td class="colleft">Test Type</td>
            <td class="colrite"><table>
                <tr>
                  <td><input type="radio" name="test_type" value="mail" <?php echo ( $_POST['test_type'] == 'mail') ? 'checked' : ''; ?>></td>
                  <td>Mail()</td>
                </tr>
                <tr>
                  <td><input type="radio" name="test_type" value="sendmail" <?php echo ( $_POST['test_type'] == 'sendmail') ? 'checked' : ''; ?>></td>
                  <td>Sendmail</td>
                </tr>
                <tr>
                  <td><input type="radio" name="test_type" value="qmail" <?php echo ( $_POST['test_type'] == 'qmail') ? 'checked' : ''; ?>></td>
                  <td>Qmail</td>
                </tr>
                <tr>
                  <td><input type="radio" name="test_type" value="smtp" <?php echo ( $_POST['test_type'] == 'smtp') ? 'checked' : ''; ?>></td>
                  <td>SMTP</td>
                </tr>
              </table>
            </td>
          </tr>
        </table>
        If SMTP test:<br />
        <table border="1" width="450" cellspacing="0" cellpadding="5" style="border-collapse: collapse;" bgcolor="#FFFFFF">
          <tr>
            <td class="colleft">SMTP Debug ?</td>
            <?php $value = ( $_POST['smtp_debug'] != '' ) ? $_POST['smtp_debug'] : $CFG['smtp_debug']; ?>
            <td class="colrite"><select size="1" name="smtp_debug">
              <option <?php echo ( $value == '0') ? 'selected' : ''; ?> value="0">0 - Disabled</option>
              <option <?php echo ( $value == '1') ? 'selected' : ''; ?> value="1">1 - Errors and Messages</option>
              <option <?php echo ( $value == '2') ? 'selected' : ''; ?> value="2">2 - Messages only</option>
              </select></td>
          </tr>
          <tr>
            <td class="colleft">SMTP Server</td>
            <?php $value = ( $_POST['smtp_server'] != '' ) ? $_POST['smtp_server'] : $CFG['smtp_server']; ?>
            <td class="colrite"><input type="text" name="smtp_server" value="<?php echo $value; ?>" style="width:99%;"></td>
          </tr>
          <tr>
            <td class="colleft">SMTP Port</td>
            <?php $value = ( $_POST['smtp_port'] != '' ) ? $_POST['smtp_port'] : $CFG['smtp_port']; ?>
            <td class="colrite"><input type="text" name="smtp_port" value="<?php echo $value; ?>" style="width:99%;"></td>
          </tr>
          <tr>
            <td class="colleft">SMTP Authenticate ?</td>
            <?php $value = ( $_POST['smtp_authenticate'] != '' ) ? $_POST['smtp_authenticate'] : $CFG['smtp_authenticate']; ?>
            <td class="colrite"><input type="checkbox" name="smtp_authenticate" <?php if ($value!=''){ echo "checked";} ?> value="<?php echo $value; ?>"></td>
          </tr>
          <tr>
            <td class="colleft">Authenticate Username</td>
            <?php $value = ( $_POST['authenticate_username'] != '' ) ? $_POST['authenticate_username'] : $CFG['smtp_username']; ?>
            <td class="colrite"><input type="text" name="authenticate_username" value="<?php echo $value; ?>" style="width:99%;"></td>
          </tr>
          <tr>
            <td class="colleft">Authenticate Password</td>
            <?php $value = ( $_POST['authenticate_password'] != '' ) ? $_POST['authenticate_password'] : $CFG['smtp_password']; ?>
            <td class="colrite"><input type="password" name="authenticate_password" value="<?php echo $value; ?>" style="width:99%;"></td>
          </tr>
        </table>
      </td>
    </tr>
  </table>
  <br />
  Test will include two attachments, plus one of the attachments is used as an inline graphic in the message body.<br />
  <br />
  <input type="submit" value="Submit" name="submit">
  </form>
  <?php
}

?>
